internal/gcsx/inactive_timeout_reader_test.go

```diff
@@ -0,0 +1,310 @@
+		Generation: s.object.Generation,
+		Range: &gcs.ByteRange{
+			Start: uint64(startOffset),
+			Limit: s.object.Size,
+		},
+		ReadCompressed: s.object.HasContentEncodingGzip(),
+		ReadHandle:     s.readHandle,
+	}
+	s.mockBucket.On("NewReaderWithReadHandle", mock.Anything, readObjectRequest).Return(s.initialFakeReader, nil).Times(1)
+
+	var err error
+	// Use NewInactiveTimeoutReader directly as NewStorageReaderWithInactiveTimeout is deprecated.
+	s.reader, err = NewInactiveTimeoutReaderWithClock(s.ctx, s.mockBucket, s.object, s.readHandle, startOffset, int64(s.object.Size), s.timeout, s.simulatedClock)
+	time.Sleep(5 * time.Millisecond) // Allow time to schedule and create a timer.
+	s.Require().NoError(err)
+	s.Require().NotNil(s.reader)
+}
+
+func (s *InactiveTimeoutReaderTestSuite) Test_NewInactiveTimeoutReader_InitialReadError() {
+	s.initialData = make([]byte, 100) // Size doesn't matter here
+	s.object = &gcs.MinObject{Name: "fail-object", Generation: 456, Size: 100}
+	s.timeout = 100 * time.Millisecond
+	initialErr := errors.New("initial connection failed")
+	// Expect the initial NewReader call to fail
+	s.mockBucket.On("NewReaderWithReadHandle", mock.Anything, mock.MatchedBy(func(req *gcs.ReadObjectRequest) bool {
+		return req.Name == s.object.Name &&
+			req.Generation == s.object.Generation &&
+			req.Range.Start == 0 &&
+			req.Range.Limit == 100
+	})).Return(nil, initialErr).Once()
+
+	_, err := NewInactiveTimeoutReader(s.ctx, s.mockBucket, s.object, []byte{}, 0, 100, s.timeout)
+
+	s.Error(err)
+	s.ErrorIs(err, initialErr) // Should be the exact error from the bucket
+}
+
+func (s *InactiveTimeoutReaderTestSuite) Test_NewInactiveTimeoutReader_ZeroTimeoutError() {
+	s.initialData = []byte("zero timeout")
+	s.timeout = 0 * time.Millisecond // Zero timeout
+
+	_, err := NewInactiveTimeoutReader(s.ctx, s.mockBucket, s.object, []byte{}, 0, 100, s.timeout)
+
+	s.Error(err)
+	s.Equal(ErrZeroInactivityTimeout, err)
+}
+
+func (s *InactiveTimeoutReaderTestSuite) Test_Read_SuccessfulWithinTimeout() {
+	// Test initial read.
+	{
+		s.initialData = []byte("hello world")
+		s.timeout = 100 * time.Millisecond
+		s.setupReader(0) // Initial read from offset 0
+		buf1 := make([]byte, 5)
+
+		n1, err1 := s.reader.Read(buf1)
+
+		s.NoError(err1)
+		s.Equal(5, n1)
+		s.Equal("hello", string(buf1[:n1]))
+	}
+
+	// Test no reader closure within timeout.
+	{
+		buf2 := make([]byte, 6)
+		s.simulatedClock.AdvanceTime(s.timeout / 2)
+		// Allow some time to routine incase timer fired in half timeout.
+		time.Sleep(5 * time.Millisecond)
+
+		n2, err2 := s.reader.Read(buf2)
+
+		inactiveReader := s.reader.(*inactiveTimeoutReader)
+		s.True(inactiveReader.isActive)
+		s.NoError(err2)
+		s.Equal(6, n2)
+		s.Equal(" world", string(buf2[:n2]))
+	}
+
+	// EOF
+	{
+		buf3 := make([]byte, 6)
+
+		n3, err3 := s.reader.Read(buf3)
+
+		s.Same(io.EOF, err3)
+		s.Equal(0, n3)
+	}
+}
+
+func (s *InactiveTimeoutReaderTestSuite) Test_Read_ReconnectFails() {
```

Comment:
Similar for this - let's break it down. Also, we don't need to verify the initial-read works in all tests btw.